rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/validation/RpkiRepositoryValidationService.java

```diff
@@ -429,15 +428,17 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
         commitPendingObjects.run();
     }
 
-    private int storePendingObjects(Tx.Write tx, RsyncRepositoryValidationRun validationRun, List<Pair<String,byte[]>> pendingObjects) {
+    private int storePendingObjects(RsyncRepositoryValidationRun validationRun, List<Pair<String,byte[]>> pendingObjects) {
         AtomicInteger counter = new AtomicInteger();
 
         // Parsing RPKI objects is CPU bound, so do this with any available threads
```

Comment:
I'd like if it we document somewhere that `parallelStream` while holding a transaction is a footcannon on machines with a low number of cores.

Reply:
We should probably add some design notes documentation file with stuff like this. The bigger foot cannon (leg cannon?) is probably the other way around: running Xodus transactions from within the fork-join pool w/o managed blocking.